optaplanner-persistence/optaplanner-persistence-jackson/src/main/java/org/optaplanner/persistence/jackson/api/score/constraint/ConstraintMatchTotalJacksonJsonDeserializer.java

```diff
@@ -0,0 +1,52 @@
+package org.optaplanner.persistence.jackson.api.score.constraint;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import com.fasterxml.jackson.core.JsonParseException;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+import com.fasterxml.jackson.databind.JsonMappingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.type.TypeFactory;
+import org.optaplanner.core.api.score.Score;
+import org.optaplanner.core.api.score.constraint.ConstraintMatch;
+import org.optaplanner.core.api.score.constraint.ConstraintMatchTotal;
+import org.optaplanner.persistence.jackson.api.score.DeserializationUtils;
+
+public class ConstraintMatchTotalJacksonJsonDeserializer extends JsonDeserializer<ConstraintMatchTotal> {
+
+    private ConstraintMatchJacksonJsonDeserializer constraintMatchDeserializer;
+
+    public ConstraintMatchTotalJacksonJsonDeserializer() {
+        constraintMatchDeserializer = new ConstraintMatchJacksonJsonDeserializer();
+    }
+
+    @Override
+    public ConstraintMatchTotal deserialize(JsonParser parser, DeserializationContext context) throws IOException {
+        ObjectMapper mapper = new ObjectMapper();
```

Comment:
Isn't creating a new ObjectMapper as part of a deserialiation problematic, because of the bootstrap cost and because it doesn't inherit the original's ObjectMapper's configuration? For example, one of the justification objects might be a Foo object for which on the origional ObjectMapper a special (de)serializer was configured.

Reply:
Thought I changed all of them, forgot I also used the ObjectMapper techique outside of DeserializationUtils; thanks, fixing now.